src/java/org/apache/cassandra/FilterExperiment.java

```diff
@@ -147,10 +150,67 @@ private static Iterator<Cell> iterator(ColumnFamily columnFamily) {
                                 Predicates.not(columnFamily.inOrderDeletionTester()::isDeleted));
     }
 
+    private static Optional<Cell> getNext(Iterator<Cell> iterator)
+    {
+        return iterator.hasNext() ? Optional.ofNullable(iterator.next()) : Optional.empty();
```

Comment:
If we're mapping null cells to empty, can we avoid optional allocation overhead and just return `@Nullable`?